zulip_bots/zulip_bots/bots/link_shortener/link_shortener.py

```diff
@@ -0,0 +1,63 @@
+import re
+import requests
+
+class LinkShortenerHandler(object):
+    '''A Zulip bot that will shorten URLs ("links") in a conversation using the
+    goo.gl URL shortener.
+    '''
+
+    def usage(self):
+        return (
+            'Mention the link shortener bot in a conversation and then enter '
+            'any URLs you want to shorten in the body of the message. \n\n'
+            '`key` must be set in `link_shortener.conf`.')
+
+    def initialize(self, bot_handler):
+        self.config_info = bot_handler.get_config_info('link_shortener')
+
+    def handle_message(self, message, bot_handler):
+        REGEX_STR = (
+            '('
+            '(?:http|https):\/\/' # This allows for the HTTP or HTTPS protocol.
+            '[^"<>#%\{\}|\\^~[\]` ]+' # This allows for any character except
+                                      # for certain non-URL-safe ones.
+            ')'
+        )
+
+        content = message['content']
+        link_matches = re.findall(REGEX_STR, content)
+
+        shortened_links = [self.shorten_link(link) for link in link_matches]
+        link_pairs = [
+            (link_match + ': ' + shortened_link)
+                for link_match, shortened_link
+                in zip(link_matches, shortened_links)
+                if shortened_link != ''
+        ]
+        final_response = '\n'.join(link_pairs)
+
+        if final_response == '':
+            return
+
+        bot_handler.send_reply(message, final_response)
+
+    def shorten_link(self, long_url):
+        '''Shortens a link using goo.gl Link Shortener and returns it, or
+        returns an empty string if something goes wrong.
+
+        Parameters:
+            long_url (str): The original URL to shorten.
+        '''
+
+        body = { 'longUrl': long_url }
+        params = { 'key': self.config_info['key'] }
+
+        request = requests.post(
+            'https://www.googleapis.com/urlshortener/v1/url',
+            json=body,
+            params=params
+        )
+
+        return request.json().get('id', '').
```

Comment:
There is a syntax error on this line. See the trailing '.'.

Reply:
OK, I fixed it in commit 17ac82d3e930700e2d26d7bdf63d407f5cb99ed8. Thanks!